src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -9258,21 +9258,134 @@ void M(S s)
                 Diagnostic(ErrorCode.ERR_FixedNeeded, "&(s_f.Buf[0])").WithLocation(17, 18));
         }
 
+        [Fact, WorkItem(39632, "https://github.com/dotnet/roslyn/issues/39632")]
+        public void CanUseEmptyInitialiserWithFixedSizeBuffer()
+        {
+            var source = @"
+unsafe struct S
+{
+    public fixed double V[3];
+
+    static void M()
+    {
+        var s = new S { V = {} };
+    }
+}";
+            CreateCompilation(source, options: TestOptions.UnsafeDebugDll).VerifyDiagnostics();
+        }
+
         [Fact, WorkItem(39632, "https://github.com/dotnet/roslyn/issues/39632")]
         public void CanUseIndexInitialiserWithFixedSizeBuffer()
         {
             var source = @"
+using System;
+
 unsafe struct S
 {
     public fixed double V[3];
 
+    static void Main()
+    {
+        var s = new S { V = 
+        {
+            [0] = 0,
+            [1] = 1,
+            [2] = 2,
+        } };
+        
+        Console.Write(s.V[0]);
+        Console.Write(s.V[1]);
+        Console.Write(s.V[2]);
+    }
+}";
+            CompileAndVerify(
+                source,
+                options: TestOptions.UnsafeReleaseExe,
+                expectedOutput: "012",
+                verify: Verification.Fails);
+        }
+
+        [Fact, WorkItem(39632, "https://github.com/dotnet/roslyn/issues/39632")]
+        public void CanUseIndexInitialiserWithFixedSizeBufferNestedInClass()
+        {
+            var source = @"
+using System;
+
+unsafe struct S
+{
+    public fixed double V[3];
+
+    static void Main()
+    {
+        var c = new C { S = new S { V = 
```

Comment:
Do we have a test that shows that the following fails: `new S { V = { 0, 1, 2 } }`

Reply:
Yes in Unsafe tests (Semantic instead of CodeGen tests)